src/AzureAuth/CommandMain.cs

```diff
@@ -394,35 +396,36 @@ private int GetToken()
 
                     try
                     {
-                        result = authFlow.GetTokenAsync().Result;
+                        results = authFlowExecutor.GetTokenAsync().Result;
+                        succeededResult = results.FirstOrDefault(result => result.Success == true);
                     }
                     finally
                     {
                         mutex.ReleaseMutex();
                     }
                 }
 
-                // But what if result is null? The compiler cannot ensure it won't be (yet).
-                this.eventData.Add("error_list", ExceptionListToStringConverter.Execute(result.Errors));
+                var errors = results.SelectMany(result => result.Errors).ToList();
+                this.eventData.Add("error_list", ExceptionListToStringConverter.Execute(errors));
 
-                if (!result.Success)
+                if (succeededResult == null)
                 {
                     this.logger.LogError("Authentication failed. Re-run with '--verbosity debug' to get see more info.");
                     return 1;
                 }
 
-                this.eventData.Add("auth_type", $"{result.TokenResult.AuthType}");
+                this.eventData.Add("auth_type", $"{succeededResult.TokenResult.AuthType}");

```

Comment:
Is it guarenteed that a non-null TokenResult will always have a non-null AuthType? Or should we do a null check here?

Reply:
from the unit tests it looks like it will always have a default `silent` AuthType. Figuring it out how.

Reply:
Just learnt that the default value of an enum property is always 0. So AuthType for TokenResult is Silent by default.

Reply:
There is an interesting divergence here - previously we were not using `null` to indicate failed auth. We designed AuthFlowResult to represent a failed auth attempt, while itself not being null.

This is why it has the Success property. By now that we have a list of them, and are searching for one with success, if we don't find one, we are left with null. I almost want to say the final result should just be the Last authFlowResult in the list. 
I think this is fine. Given you can run the CLI in such a way that no auth flows are attempted, in which case, the result list will be empty.

Right enums by default cannot be null, Though we do have ensure ourselves that it's set correctly. (Which can assume is done here)